tests/flow/tests_dag.py

```diff
@@ -771,6 +771,7 @@ def run():
             # assert that result tensor exists
             ret = con.execute_command("EXISTS {}".format(result_tensor_keyname))
             env.assertEqual(ret, 1)
+            t.join()
```

Comment:
Don't you want to join even before(line 763)? Is there a way to verify that batching actually occurred?